libs/blocks/preflight/accessibility/check-aria-labels.js

```diff
@@ -0,0 +1,98 @@
+import { getUniqueSelector } from './helper.js';
+
+export default function checkAriaLabels(elements = [], config = {}) {
+  const { checks = [] } = config;
+  if (!checks.includes('aria-labels')) return [];
+  const violations = [];
+  const targetElements = elements.filter((el) => {
+    const tag = el.tagName.toLowerCase();
+    const hasConButtonClass = el.classList?.contains('con-button');
+    const isRoleButton = el.getAttribute('role') === 'button';
+    const isAnchorCTA = tag === 'a' && (hasConButtonClass || isRoleButton);
+    return hasConButtonClass || tag === 'button' || isAnchorCTA || isRoleButton;
+  });
+
+  const missingAriaLabel = targetElements.filter((el) => !el.hasAttribute('aria-label') && !el.textContent.trim());
+  missingAriaLabel.forEach((el) => {
+    violations.push({
+      description: 'Element is missing the aria-label attribute.',
+      impact: 'serious',
+      id: 'aria-labels',
+      help: 'Add the aria-label attribute that describes the element.',
+      helpUrl: 'https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA6',
+      nodes: [{
+        target: [getUniqueSelector(el)],
+        html: el.outerHTML,
+      }],
+    });
+  });
+
+  const fillerWords = ['the', 'about', 'with', 'for', 'and', 'or', 'but', 'if', 'then', 'else', 'when', 'where', 'how', 'why', 'who', 'what', 'which', 'that', 'this', 'these', 'those'];
```

Comment:
That wouldn't work across locales right? Seems a bit weird to have a list of magic words